src/cnaas_nms/api/device.py

```diff
@@ -99,7 +99,7 @@ def delete(self, device_id):
         """ Delete device from ID """
         json_data = request.get_json()
 
-        if 'factory_default' in json_data:
+        if json_data is not None and 'factory_default' in json_data:
```

Comment:
remove "is not None" to make it more readable, just "if json_data and ..." is fine